cmd/ggit/main.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/go-git/go-git/v5/utils/trace"
+	"github.com/spf13/cobra"
+	gossh "golang.org/x/crypto/ssh"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "ggit [<args>] <command>",
+	Short: "ggit is a Git CLI that uses go-git as its backend.",
+	RunE: func(cmd *cobra.Command, _ []string) error {
+		return cmd.Usage()
+	},
+	DisableFlagsInUseLine: true,
+}
+
+// envToTarget maps what environment variables can be used
+// to enable specific trace targets.
+var envToTarget = map[string]trace.Target{
+	"GIT_TRACE":             trace.General,
+	"GIT_TRACE_PACKET":      trace.Packet,
+	"GIT_TRACE_SSH":         trace.SSH,
+	"GIT_TRACE_PERFORMANCE": trace.Performance,
+}
+
+func init() {
+	// Set up tracing
+	var target trace.Target
+	for k, v := range envToTarget {
+		if ok, _ := strconv.ParseBool(os.Getenv(k)); ok {
+			target |= v
+		}
+	}
+	trace.SetTarget(target)
+}
```

Comment:
This should run automatically from `go-git`, as all `init()` blocks from imported dependencies are called.

```suggestion
```

Reply:
IMO, this shouldn't be in `go-git` the library. Instead, tracing should be explicitly enabled in applications that use `go-git` like this cli

Reply:
I see the allure to that - overall the consumption of envars is something that IMO need to be refactored, so that it provides less surprises for the `go-git` library users - while still providing the same capabilities as the upstream CLI.